invenio_oauthclient/handlers/base.py

```diff
@@ -46,14 +47,63 @@
 )
 
 
+def _role_needs_update(role_obj, new_role_dict):
+    """Checks if role needs to be updated."""
+    if role_obj.name != new_role_dict.get("name") or role_obj.description != new_role_dict.get("description"):
+        return True
+    return False
+
+
+def _create_or_update_groups(account_groups):
+    """Creates the roles based on the groups provided."""
+    roles_id = []
+    for group in account_groups:
+        existing_role = current_datastore.find_role_by_id(group["id"])
```

Comment:
I thought I have added a comment here, but I don't see it.
I think that we should avoid this 2 checks, for performance reasons. This happens during the HTTP request of the login: for users with high number of groups, it adds extra delays.

Can we try to insert and catch the failure instead?